### DISCUSSION TITLE
Ethernet Not Working on Intel Ultra 9 185H with Realtek r8126 Driver on Linux Mint Wilma 22

### DISCUSSION THREAD

**Discussion by sueknock** [linuxmint/discussions#603] [Questions]
I am experiencing an issue with Ethernet connectivity on my MinisForum AtomMan Intel Ultra 9 running Linux Mint Wilma version 22. I have two Ethernet ports (enp86s0 and enp87s0), and both are showing <NO-CARRIER> status despite multiple cables and restarts of both the PC and router. I have installed the Realtek r8126 drivers (version 10.013.00-NAPI) but still cannot get Ethernet to work. 

Below is the relevant system information:
System Information:
Kernel version: 6.8.0-41-generic

Network interfaces:
enp86s0: <NO-CARRIER,BROADCAST,MULTICAST,UP>
enp87s0: <NO-CARRIER,BROADCAST,MULTICAST,UP>
WiFi interface: wlp88s0f0 is functioning normally.

Realtek driver status: r8126 196608 0

Ethernet controller information:
56:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. Device 8126 (rev 01)
57:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. Device 8126 (rev 01)

dmesg logs related to Ethernet:
[0.828564] r8126 Ethernet controller driver 10.013.00-NAPI loaded
[0.845425] r8126 Ethernet controller driver 10.013.00-NAPI loaded

Despite this, both 5G Realtek Ethernet ports show as not connected (despite checking multiple cables and router). Any guidance on resolving this issue would be greatly appreciated!

Let me know if more information is needed.

**Comment by rcalixte** [comment]
The Core Ultra 9 185H CPU was released in late 2023. It may take a few kernel iterations before there is full support upstream. Version 6.11 of the kernel should be available soon in the Ubuntu repositories and offer improved support as well. Your options are:

* waiting for the newer kernel to appear as an update
* manually trying to install a newer kernel (via PPA or other source)
* switching to a different distro that offers a newer kernel by default (the nuclear option)

The bad news is that this a known caveat sometimes from using newer hardware with Linux. The good news is that the time to a resolution is shorter than it used to be historically.

**Comment by sueknock** [comment]
Thank you for your detailed response.  Just a quick follow up for my clarification. The problem seems to be with my Realtek 5GBe Ethernet adapters, not the Intel Core Ultra 9 CPU right? Is it not possible to get newer drivers from Realtek easier than waiting for Ubuntu kernel update to 6.11? How will that solve a Realtek Ethernet Driver issue? Thanks. On Sep 4, 2024, at 8:59 PM, Rick Calixte ***@***.***> wrote:﻿
The Core Ultra 9 185H CPU was released in late 2023. It may take a few kernel iterations before there is full support upstream. Version 6.11 of the kernel should be available soon in the Ubuntu repositories and offer improved support as well. Your options are:

waiting for the newer kernel to appear as an update
manually trying to install a newer kernel (via PPA or other source)
switching to a different distro that offers a newer kernel by default (the nuclear option)

The bad news is that this a known caveat sometimes from using newer hardware with Linux. The good news is that the time to a resolution is shorter than it used to be historically.

**Reply by rcalixte** [reply]
> The problem seems to be with my Realtek 5GBe Ethernet adapters, not the Intel Core Ultra 9 CPU right? Is it not possible to get newer drivers from Realtek easier than waiting for Ubuntu kernel update to 6.11? How will that solve a Realtek Ethernet Driver issue?

The drivers are included in the upstream kernel. An example is [here](https://linux-hardware.org/?id=pci:10ec-8126-1458-e000). You can try to find and install drivers manually if they are available online with instructions but if you don't have experience with this, it would be easier to wait. The CPU is often a good estimator of when you might expect to have good to full support by the kernel for your hardware. It's not perfect but hardware is typically released in pairings which is why it is generally easier for estimation purposes.
